src/test/java/com/google/research/bleth/services/BoardParserTest.java

```diff
@@ -0,0 +1,137 @@
+package com.google.research.bleth.services;
+
+import com.google.appengine.repackaged.com.google.gson.Gson;
+import com.google.common.collect.ArrayTable;
+import com.google.research.bleth.simulator.Agent;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.Location;
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BoardParserTest extends TestCase {
+
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+
+    @Mock
+    private Agent firstAgent;
+    @Mock
+    private Agent secondAgent;
+
+    public ArrayTable<Integer, Integer, ArrayList<String>> emptyStringListTable() {
```

Comment:
Prefer to move helpers below the actual testcases: the testcases are more important to a reader of this class.

Also - helpers should be private. (Always minimize visibility of all elements in your code...)